investigate/investigate.py

```diff
@@ -232,5 +237,45 @@ def search(self, pattern, start=None, limit=None, include_category=None):
 
         return self.get_parse(uri, params)
 
+    def samples(self, anystring, limit=None, offset=None, sortby=None):
+        '''Return an object representing the samples identified by the input domain, IP, or URL'''
 
+        uri = self._uris['samples'].format(anystring)
+        params = {'limit': limit, 'offset': offset, 'sortby': sortby}
 
+        return self.get_parse(uri, params)
+
+    def sample(self, hash, limit=None, offset=None):
+        '''Return an object representing the sample identified by the input hash, or an empty object if that sample is not found'''
+
+        uri = self._uris['sample'].format(hash)
+        params = {'limit': limit, 'offset': offset}
+
+        return self.get_parse(uri, params)
+
+    def sample_artifacts(self, hash, limit=None, offset=None):
+        '''
+            Return an object representing artifacts associated to an input hash
```

Comment:
Minor bit, but this should say `associated with an input hash`, similar minor grammar issue on the the other endpoints below.